### PR TITLE
HPCC-32434 Fix constructor order warnings

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
(Note - I'm not sure I've ever seen a situation where this particular warning helped highlight a bug, so another approach would be to disable the warning...)

**Review by ghalliday — APPROVED** [review]
All the changes look good.  Slight hesitation about merging into master rather than 9.8.x - will it result in lots of upmerge clashes?
I suspect not because any new code is likely to use initialisers on members, and not expecting lots of changes.  There is also the possibility that new code in 9.6. or 9.8 will fail when upmerged, so need to look out for that.


It is also safer to merge to master in case it causes a regression - although I have checked for examples that pass this

**Review by ghalliday — CHANGES_REQUESTED** [review]
MacOs build failure is another example that needs reordering.

**Comment by richardkchapman** [comment]
Not sure why some changes were missing - I suspect they were included in earlier commits and thus lost when I split into a separate PR.

I have included a few extra changes and repushed.

File: cmake_modules/commonSetup.cmake [hunk 1]
```diff
@@ -370,7 +370,8 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
       if (CLANG_VERSION VERSION_GREATER 3.6 OR CLANG_VERSION VERSION_EQUAL 3.6 OR APPLE_CLANG_VERSION VERSION_GREATER 6.0)
         SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pointer-bool-conversion")
       endif()
-    endif()
+      SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=reorder-ctor")
+      endif()
```
Comment (ghalliday) [inline cmake_modules/commonSetup.cmake:374]: @richardkchapman indentation problem, but will merge as-is.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**